monai/deploy/packager/util.py

```diff
@@ -42,7 +42,11 @@ def verify_base_image(base_image: str) -> str:
         str: returns string identifier of the dockerfile template to build MAP
         if valid base image provided, returns empty string otherwise
     """
-    valid_prefixes = {"nvcr.io/nvidia/cuda": "ubuntu", "nvcr.io/nvidia/pytorch": "pytorch"}
+    import torch
+    if "AMD" not in torch.cuda.get_device_name(0):
```

Comment:
This introduces a limitation on the Packager in that a GPU must be present on the local host, and its type determines if  what the built container is for AMD of not. This means,
1. Cannot build a container with AMD base on a host with Nvidia GPU
2. The script fails when none GPU is visible; this can be tested with setting Nvidia GPU to invisible. 
```
     import os
     os.environ['CUDA_VISIBLE_DEVICES'] = ''
     import torch
     torch.cuda.is_available()  # -> False
     torch.cuda.get_device_name(0)  # -> RuntimeError: No CUDA GPUs are available
```
    Also, `torch.cuda.get_device_name()` causes the script to hang when GPU not available 

So, at least a try/except block is needed to prevent the failure, and make known an MAP targeting AMD GPU cannot only be built on host with an AMD GPU.